root/static/scripts/edit/URLCleanup.js

```diff
@@ -915,6 +915,40 @@ const CLEANUPS: CleanupEntries = {
       return url.replace(/^(https:\/\/archive\.org\/details\/[A-Za-z0-9._-]+)\/$/, '$1');
     },
   },
+  'audiomack': {
+    match: [new RegExp('^(https?://)?([^/]+\\.)?audiomack\\.com/', 'i')],
+    restrict: [LINK_TYPES.streamingfree],
+    clean: function (url) {
+      url = url.replace(/^https?:\/\/(www.)?audiomack.com\//, 'https://audiomack.com/');
+      return url;
+    },
+    validate: function (url, id) {
+      const m = /^https:\/\/audiomack\.com\/([a-zA-Z0-9_-]+)(?:\/(album|song)\/[a-zA-Z0-9_-]+\/?)?$/.exec(url);
```

Comment:
Are names for artist, album, and song assumed to be written with these symbols only or is there any reference about that?

Reply:
Assumed. I didn't find any reference (although I might have missed it) so I checked a fair amount of URLs. A-Z is just in case someone uppercases them for reasons, and I'm honestly not even sure they support _. But this is a common choice in our URLs so I thought I'd go with it for now :)

Reply:
Maybe just mention in commit message that it is your best guess so we know what we had if this breaks later on.

Otherwise it is also doable to be even less restrictive: `[^/?&#]+`